lib/hashes.md

```diff
@@ -5,40 +5,68 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A Hash is a dictionary-like collection of unique keys and their values. They are similar to Arrays, but where an Array uses integers as its index, a Hash allows you to use any object type.
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+As stated above, Hashes can hold any type of object data. For example: grades = { "Jane Doe" => 10, "Jim Doe" => 6 } 
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+They are similar in the fact that they both hold indexes of information like a library. The difference is that hashes can hold many more types of data than an array can.
+
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+you would use the command hash.[key]. An example would be: 
+
+elements = {"base"=>"foundation", "pedestal"=>"base"}
+  puts elements["base"]
+  
+  and that will bring back the result "foundation"
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+We would replace it by performing this operation:
+hash = {:item1 => 1}
+hash[:item2] = 2
 
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+I believe it is similar to that of an array. You would take your elements inside your hash and bundle them under another name, such as Contacts.
+You would then type in whatever command you wanted to go with it, like Contacts.delete[David] and that would remove David from out list.
```

Comment:
Okay I think I have it this time around. I appreciate the help and let me know if this looks right!

scores = {"Tyler"=> 500, "Beth" => 102, "Crystal" => 429}

scores.each do | name, points |
  puts "#{name} scores #{points} points"
end